aspnetcore/fundamentals/configuration.md

```diff
@@ -5,162 +5,981 @@ ms.author: riande
+   }
+   ````
 
-![image](configuration/_static/config-console.png)
+<a name=options-example></a>
 
-<a name=options-config-objects></a>
+In the following code, the JSON configuration provider is enabled and the `MyOptions` class is added to the service container. The `MyOptions` class is bound to configuration.
 
-## Using Options and configuration objects
+[!code-csharp[Main](configuration/sample/src/UsingOptions/Startup.cs?name=snippet1)]
 
-The options pattern enables using custom options classes to represent a group of related settings. A class needs to have a public read-write property for each setting and a constructor that does not take any parameters (e.g. a default constructor) in order to be used as an options class.
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "source": "/Users/shirhatti/src/Docs/aspnet/fundamentals/configuration/sample/src/UsingOptions/Startup.cs", "highlight_args": {"linenostart": 1, "hl_lines": [5, 6, 7, 8, 10, 13, 15, 17, 18, 20, 21]}, "names": []} -->
 
-It's recommended that you create well-factored settings objects that correspond to certain features within your application, thus following the [Interface Segregation Principle (ISP)](http://deviq.com/interface-segregation-principle/) (classes depend only on the configuration settings they use) as well as [Separation of Concerns](http://deviq.com/separation-of-concerns/) (settings for disparate parts of your app are managed separately, and thus are less likely to negatively impact one another).
+````c#
 
-A simple `MyOptions` class is shown here:
+   public class Startup
+   {
+// RM
+       public Startup(IHostingEnvironment env)
+       {
+           // Set up configuration sources.
+           var builder = new ConfigurationBuilder()
+               .SetBasePath(env.ContentRootPath)
+               .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true);
 
-[!code-csharp[Main](configuration/sample/src/UsingOptions/Models/MyOptions.cs?range=3-7)]
+           Configuration = builder.Build();
+       }
 
-Options can be injected into your application using the `IOptions<TOptions>` accessor service. For example, the following [controller](../mvc/controllers/index.md)  uses `IOptions<MyOptions>` to access the settings it needs to render the `Index` view:
+       public IConfigurationRoot Configuration { get; set; }
 
-[!code-csharp[Main](../fundamentals/configuration/sample/src/UsingOptions/Controllers/HomeController.cs?highlight=3,5,8&range=9-20)]
+       public void ConfigureServices(IServiceCollection services)
+       {
+           // Adds services required for using options.
+           services.AddOptions();
 
->[!TIP]
-> Learn more about [Dependency Injection](dependency-injection.md).
+           // Register the ConfigurationBuilder instance which MyOptions binds against.
+           services.Configure<MyOptions>(Configuration);
```

Comment:
`Configuration` is not a `ConfigurationBuilder` instance but an `IConfiguration` (more specifically an `IConfigurationRoot`, but that is probably not relevant). 

Consider using the overload of `Configure<TOptions>()` that also takes a string key to select a section of the configuration to bind to the options. Binding the whole configuration to a single options object should be rare in real apps.